framework/SwiftPlot/Renderer.swift

```diff
@@ -5,7 +5,8 @@ public protocol Renderer: AnyObject{
 	var plotDimensions: PlotDimensions { get set }
 
 	func drawRect(topLeftPoint p1: Point, topRightPoint p2: Point, bottomRightPoint p3: Point, bottomLeftPoint p4: Point, strokeWidth thickness: Float, strokeColor: Color)
-	func drawSolidRect(topLeftPoint p1: Point, topRightPoint p2: Point, bottomRightPoint p3: Point, bottomLeftPoint p4: Point, fillColor: Color)
+	func drawSolidRect(topLeftPoint p1: Point, topRightPoint p2: Point, bottomRightPoint p3: Point, bottomLeftPoint p4: Point, fillColor: Color, hatchPattern: Series.hatching)
+	func drawSolidRectTransformed(topLeftPoint p1: Point, topRightPoint p2: Point, bottomRightPoint p3: Point, bottomLeftPoint p4: Point, fillColor: Color, hatchPattern: Series.hatching)
```

Comment:
What's the difference between `drawSolidRect` and `drawSolidRectTransformed`? Some comments on the methods in this protocol would be helpful.

Is there some way that you could do everything you need with only one function, so that renderer implementations don't have to implement as much logic?

Reply:
All the functions with the Transformed suffix use a coordinate system with the shifted origin which is shifted to the bottom left of the graph border. Maybe Transformed isn't the best suffix to indicate that. 

I can write all the code in one function with a boolean parameters decide if we want to shift the origin or not, if that seems alright?